src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/OnceInitializedOnceDisposedUnderLockAsync.cs

```diff
@@ -112,29 +114,29 @@ private async Task<T> ExecuteUnderLockCoreAsync<T>(Func<CancellationToken, Task<
         {
             Requires.NotNull(action, nameof(action));
 
-            using (var source = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, DisposalToken))
+            using var source = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, DisposalToken);
+            CancellationToken jointCancellationToken = source.Token;
+
+            try
+            {
+                T result = default!;
+                await _semaphore.ExecuteAsync(async () => { result = await action(jointCancellationToken); }, jointCancellationToken);
+
+                return result;
+            }
+            catch (Exception ex)
             {
-                CancellationToken jointCancellationToken = source.Token;
-
-                try
-                {
-                    T result = default!;
-                    await _semaphore.ExecuteAsync(async () => { result = await action(jointCancellationToken); }, jointCancellationToken);
-
-                    return result;
-                }
-                catch (Exception ex)
-                {
-                    if (!TryTranslateException(ex, cancellationToken, DisposalToken))
-                        throw;
-                }
+                if (!TryTranslateException(ex, cancellationToken, DisposalToken))
+                    throw;
             }
 
             throw Assumes.NotReachable();
         }
 
         private async Task ExecuteUnderLockCoreAsync(Func<CancellationToken, Task> action, CancellationToken cancellationToken = default)
         {
+            Requires.NotNull(action, nameof(action));
```

Comment:
Why?

Reply:
Per 890e3b9fafad516597d102454c32e25cfa79c806, makes it consistent with the other methods in this class. Either they should all be removed, or all added. I am less bullish than you on the idea that NRT means we don't need runtime checks, for now at least.